WcaOnRails/app/views/competitions/show.html.erb

```diff
@@ -1,5 +1,23 @@
 <% provide(:title, @competition.name) %>
 
-<%= render layout: 'results_nav' do %>
-  <%= render "results_table", results: @competition.winning_results, hide_pos: true, hide_round: true %>
+<%= render layout: 'nav' do %>
+  <ul class="nav nav-tabs">
+    <li><a href="#general-info" data-toggle="tab"><%=t '.show.general_info' %></a></li>
+    <% @competition.competition_tabs.each do |tab| %>
+      <li><a href="#<%= "#{tab.id}-#{tab.name.parameterize}" %>" data-toggle="tab"><%= tab.name %></a></li>
+    <% end %>
+  </ul>
+
+  <div class="tab-content" id="competition-tabs">
+    <div class="tab-pane" id="general-info">
+      <%= render layout: 'results_nav', locals: { layout_nav: false } do %>
+        <%= render "results_table", results: @competition.winning_results, hide_pos: true, hide_round: true %>
+      <% end %>
+    </div>
+    <% @competition.competition_tabs.each do |tab| %>
+      <div class="tab-pane" id="<%= "#{tab.id}-#{tab.name.parameterize}" %>">
```

Comment:
Hm. Using the anchor makes me uneasy, but I don't have the time to figure out if maybe it's brilliant. Not the most helpful comment, I know...

Anyway, if line 7 and line 18 need to be the same strings, they should be made by the same code. You could have a method on the model object or a view helper, depending on your philosophical leanings.